libraries/freertos_plus/standard/https/include/iot_https_client.h

```diff
@@ -0,0 +1,387 @@
+ *          #IOT_HTTPS_MESSAGE_TOO_LARGE if the response cannot fit in the configured 
+ *              struct IotHttpsRequestHandle.respUserBuffer.pBuffer and struct IotHttpsRequestHandle.pSyncInfo.pRespData.
+ *          #IOT_HTTPS_CONNECTION_ERROR if the connection failed.
+ *          #IOT_HTTPS_INVALID_PARAMETER if there are NULL parameters or the request is asynchronous.
+ *          #IOT_HTTPS_NETWORK_ERROR if there was an error sending the data on the network.
+ *          #IOT_HTTPS_PARSING_ERROR if there was an error parsing the HTTP response.
+ *          Please see #IotHttpsReturnCode_t for other failure codes.
+ */
+/* @[declare_https_client_sendsync] */
+IotHttpsReturnCode_t IotHttpsClient_SendSync(IotHttpsConnectionHandle_t* pConnHandle, IotHttpsRequestHandle_t reqHandle, IotHttpsResponseHandle_t * pRespHandle);
+/* @[declare_https_client_sendsync] */
+
+/**
+ * @brief Retrieve the HTTPS response status.
+ * 
+ * The HTTP response status code is contained in the status line of the response header buffer configured in 
+ * #IotHttpsRequestInfo_t.respUserBuffer. It is the first line of a standard HTTP response message. If the response 
+ * status line could not fit into #IotHttpsRequestInfo_t.respUserBuffer, then this function will return an error code. 
+ * Please see #responseUserBufferMinimumSize for information about sizing the #IotHttpsRequestInfo_t.respUserBuffer.
+ * 
+ * This routine is primarily to be used for a synchronous response. It can be used for an asynchronous response.
+ * <b> Example Synchronous Code </b>
+ * @code{c}
+ *      ...
+ *      IotHttpsClient_SendSync(&connHandle, reqHandle, &respHandle);
+ *      uint16_t status = 0;
+ *      IotHttpsClient_ReadResponseStatus(respHandle, &status);
+ *      if (status != IOT_HTTPS_STATUS_OK)
+ *      {
+ *          // Handle error server response status.
+ *      }
+ *      ...
+ * @endcode
+ * 
+ * param[in] respHandle - Unique handle representing the HTTPS response.
+ * param[out] status - Integer status returned from the server response.
+ * 
+ * @return #IOT_HTTPS_OK if the response status was successfully read into *status.
+ *         #IOT_HTTPS_INVALID_PARAMETER for NULL parameters.
+ *         #IOT_HTTPS_NOT_FOUND if the HTTP response status was not found in the header buffer.
+ *         Error code otherwise, please see #IotHttpsReturnCode_t for for failure codes.
+ */
+/* @[declare_https_client_readresponsestatus] */
+IotHttpsReturnCode_t IotHttpsClient_ReadResponseStatus(IotHttpsResponseHandle_t respHandle, uint16_t *status);
+/* @[declare_https_client_readresponsestatus] */
+
+/**
+ * @brief Retrieve the HTTPS response content length. 
+ * 
+ * If the "Content-Length" header is available in #IotHttpsRequestInfo_t.respUserBuffer, this routine extracts that 
+ * value. The "Content-Length" header could not be available because the server sent a multi-part encoded response 
```

Comment:
```
 * If the "Content-Length" header is available in #IotHttpsRequestInfo_t.respUserBuffer, this routine extracts that 
 * value. In some cases the "Content-Length" header is be available. This could be because the server sent a multi-part 
 * encoded response (For example, "Transfer-Encoding: chunked") or the "Content-Length" header was far down in the list 
 * of response  headers and could not fit into the header buffer configured in #IotHttpsRequestInfo_t.respUserBuffer. 
 * Please see #responseUserBufferMinimumSize for information about sizing the #IotHttpsRequestInfo_t.respUserBuffer.
```